src/libraries/Microsoft.Extensions.Caching.Abstractions/src/MemoryCacheExtensions.cs

```diff
@@ -99,13 +103,11 @@ public static TItem GetOrCreate<TItem>(this IMemoryCache cache, object key, Func
         {
             if (!cache.TryGetValue(key, out object result))
             {
-                ICacheEntry entry = cache.CreateEntry(key);
-                result = factory(entry);
-                entry.SetValue(result);
-                // need to manually call dispose instead of having a using
-                // in case the factory passed in throws, in which case we
-                // do not want to add the entry to the cache
```

Comment:
Does this concern no longer apply?

Reply:
Nevermind, it'll lack the `ValueHasBeenSet = true;` part, i.e. the purpose of the PR :) haven't had my coffee yet.